daml-assistant/integration-tests/src/Main.hs

```diff
@@ -138,9 +138,9 @@ noassistantTests damlDir = testGroup "no assistant"
               callProcess damlcPath ["build", "--project-root", "foobar", "--init-package-db", "yes"]
     ]
 
-packagingTests :: FilePath -> TestTree
-packagingTests tmpDir = testGroup "packaging"
-    [ testCaseSteps "Build package with dependency" $ \step -> do
+packagingTests :: TestTree
+packagingTests = testGroup "packaging"
+    [ testCaseSteps "Build package with dependency" $ \step -> withTempDir $ \tmpDir -> do
```

Comment:
What changed that requires us to use `withTempDir` here now? It looks reasonable but I don’t see why this has to change with this PR.

Reply:
we happen to use the same temporary directory for all separate packaging tests. That means it's very easy to pick up sources from other tests with unpredictable results. This, for example, happened when I changed the source field of the dependency test to "." instead of "A.daml" here.